services/elasticache/pkg/resource/replication_group/custom_update_api.go

```diff
@@ -490,6 +555,193 @@ func (rm *resourceManager) newUpdateShardConfigurationRequestPayload(
 	return res, nil
 }
 
+// describeCacheCluster provides CacheCluster object
+// per the supplied latest Replication Group Id
+// it invokes DescribeCacheClusters API to do so
+func (rm *resourceManager) describeCacheCluster(
+	ctx context.Context,
+	latest *resource,
+) (*svcsdk.CacheCluster, error) {
+	input := &svcsdk.DescribeCacheClustersInput{}
+
+	var cacheClusterId *string = nil
+	latestStatus := latest.ko.Status
+	if latestStatus.NodeGroups == nil {
+		return nil, nil
+	}
+	for _, nodeGroup := range latestStatus.NodeGroups {
+		if nodeGroup.NodeGroupMembers == nil {
+			continue
+		}
+		for _, nodeGroupMember := range nodeGroup.NodeGroupMembers {
+			if nodeGroupMember.CacheClusterID == nil {
+				continue
+			}
+			cacheClusterId = nodeGroupMember.CacheClusterID
+			break
+		}
+		if cacheClusterId != nil {
+			break
+		}
+	}
```

Comment:
Consider breaking the above out into a `getCacheClusterIDFromNodeGroups` helper function.

Also, the above assumes that all NodeGroups have the same `CacheClusterID` value. Is this always the case?

(if it is the case, one wonders why this isn't a field on the ReplicationGroup and not the NodeGroup...) :(

Reply:
Updated method name to be `getAnyCacheClusterIDFromNodeGroups`.
Any cluster Id suffice because it is used to get details about the `SecurityGroups` and `EngineVersion` data which is same for all clusters.